.ci/jenkins/lib/test-matrix.yaml

```diff
@@ -65,50 +66,62 @@ steps:
   - name: Build GPU Test Environment
     parallel: false
     run: |
+      set -x
+      # Clean any leftover nixl-ci-test containers
+      docker ps -q --filter "name=nixl-ci-test" | xargs -r docker rm -f
```

Comment:
can it cause problems for jobs running in parallel?
why needed?

Reply:
No parallel jobs can run on this Jenkins node (single executor). Updating the job definition to make this explicit: 050e499e701d2b336dbb4e0bd91618f29366e9a8

The root cause was multiple Docker containers hanging, all using the shared workspace. The pre-run cleanup ensures no leftover containers.